core/arch/arm/kernel/pseudo_ta.c

```diff
@@ -163,8 +163,9 @@ static TEE_Result pseudo_ta_enter_open_session(struct tee_ta_session *s,
 		}
 
 		res = stc->pseudo_ta->open_session_entry_point(param->types,
-								tee_param,
-								&s->user_ctx);
+							       tee_param,
+							       s->id);
```

Comment:
For socket PTA, which is the only one needing this, it's not good enough. The ID has to be globally unique in normal world. `s->id` is only unique in the callers context.

Reply:
Since https://github.com/OP-TEE/optee_os/commit/42461166ec5dc1bfd0124acd73ba53dcd962611d, the client receives an ID. For TAs, it is the TA generic shim that converts the ID into a `void *` reference. This change updates PTA to receive such session ID.

So socket PTA needs to create an ID meaningful to the PTA. `pta_invoke_test.c` does this by creating a context for each client (from open_session) and using the session_id to find back the context reference (where any client specific data could be stored).

PTA could each do this conversion. Maybe `pta_add_client()`, `pta_remove_client()`, `pta_client_s_valid()` could be made generic.

Reply:
No, the generated IDs are only unique from the caller point of view (where the `struct tee_ta_session_head *open_sessions`) is stored.

Reply:
Ok, i see how it's used by `entry_std.c` and `tee_svc.c`. Took time but now i catch what you meant by _caller point of view_.